src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/UserDefinedImplicitConversions.cs

```diff
@@ -978,5 +979,69 @@ protected UserDefinedConversionResult AnalyzeImplicitUserDefinedConversionForV6S
 
             return UserDefinedConversionResult.NoApplicableOperators(u);
         }
+
+        private Conversion AnalyzeImplicitUnionConversions(
+            BoundExpression sourceExpression,
+            TypeSymbol source,
+            TypeSymbol target,
+            ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            Debug.Assert(sourceExpression is null || Compilation is not null);
+            Debug.Assert(sourceExpression != null || (object)source != null);
+            Debug.Assert((object)target != null);
+
+            // PROTOTYPE: It might make sense to block conversions from a base class or any interface. See comments in tests.
+
+            if (target is not NamedTypeSymbol namedTarget || !namedTarget.IsUnionTypeWithUseSiteDiagnostics(ref useSiteInfo))
+            {
+                return Conversion.NoConversion;
+            }
+
+            // SPEC: Find the set of applicable constructors
```

Comment:
Consider linking to https://github.com/dotnet/csharplang/blob/e47c2d6be6b7c8b76b5c94b158e08df8048f92b2/proposals/unions.md#union-conversions here